### PR TITLE
Add webhook definitions for part of the schema

### PR DESCRIPTION
**Pull request by korycins**
I want to merge this change because it adds a description to explain which webhooks are called by specific location.
The PR covers introduces the logic for adding the webhook description to mutations and type fields.
It also covers the mutations for modules:
`account`, `app`, `attributes`, `channel`, `checkout`, `csv`, `discount`.

The mutation/field will get the webhook description as below:
```
  Triggers below webhook events:
   - SHIPPING_LIST_METHODS_FOR_CHECKOUT (sync): Optionally triggered when cached external shipping methods are invalid.
   - CHECKOUT_FILTER_SHIPPING_METHODS (sync): Optionally triggered when cached filtered shipping methods are invalid.
```
Additionaly the directive will be attached to the mutation/field:
`@webhookEventsInfo(asyncEvents: [], syncEvents: [SHIPPING_LIST_METHODS_FOR_CHECKOUT, CHECKOUT_FILTER_SHIPPING_METHODS])`

---
The declaration attached to the mutation/field looks:
```python
        webhook_events_info=[
             WebhookEventInfo(
                 type=WebhookEventSyncType.SHIPPING_LIST_METHODS_FOR_CHECKOUT,
                 description=(
                     "Optionally triggered when cached external shipping methods are "
                     "invalid."
                 ),
             ),
             WebhookEventInfo(
                 type=WebhookEventSyncType.CHECKOUT_FILTER_SHIPPING_METHODS,
                 description=(
                     "Optionally triggered when cached filtered shipping methods are "
                     "invalid."
                 ),
             ),
         ,
```

---

I will extend the rest of the modules in separate PR.

Part of #12591.

# Impact

* [ ] New migrations
* [ ] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [ ] Documentation needs to be updated

# Pull Request Checklist

* [ ] Privileged queries and mutations are either absent or guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [ ] Database migrations are either absent or optimized for zero downtime
* [ ] The changes are covered by test cases

### CODE REVIEW

File: saleor/graphql/account/bulk_mutations/customer_bulk_update.py [hunk 1]
```diff
@@ -96,6 +97,16 @@ class Meta:
         doc_category = DOC_CATEGORY_USERS
         permissions = (AccountPermissions.MANAGE_USERS,)
         error_type_class = CustomerBulkUpdateError
+        webhook_events_info = [
+            WebhookEventInfo(
+                type=WebhookEventAsyncType.CUSTOMER_UPDATED,
+                description="A new customer account was updated.",
```
Comment (maarcingebala) [inline saleor/graphql/account/bulk_mutations/customer_bulk_update.py:103]: Small change, but it's not "new" if we're updating it right?
```suggestion
                description="A customer account was updated.",
```

File: saleor/graphql/checkout/mutations/checkout_complete.py [hunk 2]
```diff
@@ -110,6 +112,56 @@ class Meta:
...
+                type=WebhookEventAsyncType.ORDER_CREATED,
+                description=("Triggered when order is created."),
+            ),
+            WebhookEventInfo(
+                type=WebhookEventAsyncType.NOTIFY_USER,
+                description=("A notification for order confirmation."),
+            ),
+            WebhookEventInfo(
+                type=WebhookEventAsyncType.NOTIFY_USER,
+                description=("A notification for order confirmed."),
```
Comment (maarcingebala) [inline saleor/graphql/checkout/mutations/checkout_complete.py:144]: These two are very similar, can we somehow explain it more explicitly?

File: saleor/graphql/checkout/mutations/checkout_delivery_method_update.py [hunk 3]
```diff
@@ -61,6 +62,16 @@ class Meta:
         )
         doc_category = DOC_CATEGORY_CHECKOUT
         error_type_class = CheckoutError
+        webhook_events_info = [
+            WebhookEventInfo(
+                type=WebhookEventSyncType.SHIPPING_LIST_METHODS_FOR_CHECKOUT,
+                description="Triggered when provided delivery method is external id.",
```
Comment (maarcingebala) [inline saleor/graphql/checkout/mutations/checkout_delivery_method_update.py:68]: Rephrase for clarity: "Triggered when provided delivery method is an external one, provided by this webhook."

File: saleor/graphql/checkout/mutations/checkout_shipping_method_update.py [hunk 4]
```diff
@@ -54,6 +55,16 @@ class Meta:
         doc_category = DOC_CATEGORY_CHECKOUT
         error_type_class = CheckoutError
         error_type_field = "checkout_errors"
+        webhook_events_info = [
+            WebhookEventInfo(
+                type=WebhookEventSyncType.SHIPPING_LIST_METHODS_FOR_CHECKOUT,
+                description="Triggered when provided delivery method is external id.",
```
Comment (maarcingebala) [inline saleor/graphql/checkout/mutations/checkout_shipping_method_update.py:61]: Same as earlier: "Triggered when provided delivery method is an external one, provided by this webhook."

File: saleor/graphql/checkout/mutations/order_create_from_checkout.py [hunk 5]
```diff
@@ -83,6 +85,56 @@ class Meta:
...
+                type=WebhookEventAsyncType.ORDER_CREATED,
+                description=("Triggered when order is created."),
+            ),
+            WebhookEventInfo(
+                type=WebhookEventAsyncType.NOTIFY_USER,
+                description=("A notification for order confirmation."),
+            ),
+            WebhookEventInfo(
+                type=WebhookEventAsyncType.NOTIFY_USER,
+                description=("A notification for order confirmed."),
```
Comment (maarcingebala) [inline saleor/graphql/checkout/mutations/order_create_from_checkout.py:117]: Same here, we should make it more clear how these two are different.

File: saleor/graphql/checkout/mutations/order_create_from_checkout.py [hunk 6]
```diff
@@ -83,6 +85,56 @@ class Meta:
...
+            ),
+            WebhookEventInfo(
+                type=WebhookEventAsyncType.ORDER_UPDATED,
```
Comment (maarcingebala) [inline saleor/graphql/checkout/mutations/order_create_from_checkout.py:120]: Missing description?

File: saleor/graphql/core/utils/__init__.py [hunk 7]
```diff
@@ -134,3 +136,24 @@ def ext_ref_to_global_id_or_error(model, external_reference):
...
+    description: Optional[str] = None
+
+
+CHECKOUT_CALCULATE_TAXES_MESSAGE = (
+    "Optionally triggered when checkout prices are expired."
+)
+
+
+def message_webhook_events(webhook_events: List[WebhookEventInfo]) -> str:
+    description = "\n\nTriggers below webhook events:"
```
Comment (maarcingebala) [inline saleor/graphql/core/utils/__init__.py:153]:
```suggestion
    description = "\n\nTriggers the following webhook events:"
```

**Review by maarcingebala — COMMENTED** [review]
@korycins It looks amazing in the schema! Added few small comments to improve some of the descriptions.

File: saleor/graphql/checkout/types.py [hunk 8]
```diff
@@ -435,10 +481,16 @@ class Checkout(ModelObjectType[models.Checkout]):
...
-    available_payment_gateways = NonNullList(
-        PaymentGateway,
+    available_payment_gateways = BaseField(
+        NonNullList(PaymentGateway),
         description="List of available payment gateways.",
         required=True,
+        webhook_events_info=[
+            WebhookEventInfo(
+                type=WebhookEventSyncType.PAYMENT_LIST_GATEWAYS,
+                description=("Fetch payment gateways available for checkout. "),
```
Comment (IKarbowiak) [inline saleor/graphql/checkout/types.py:491]: No needed parentless

File: saleor/graphql/discount/mutations/sale/sale_add_catalogues.py [hunk 9]
```diff
@@ -21,6 +23,12 @@ class Meta:
         permissions = (DiscountPermissions.MANAGE_DISCOUNTS,)
         error_type_class = DiscountError
         error_type_field = "discount_errors"
+        webhook_events_info = [
+            WebhookEventInfo(
+                type=WebhookEventAsyncType.SALE_UPDATED,
+                description=("A sale was updated."),
```
Comment (IKarbowiak) [inline saleor/graphql/discount/mutations/sale/sale_add_catalogues.py:29]: Not needed parentheses 👮‍♀️

File: saleor/graphql/discount/mutations/sale/sale_create.py [hunk 10]
```diff
@@ -70,6 +72,12 @@ class Meta:
         permissions = (DiscountPermissions.MANAGE_DISCOUNTS,)
         error_type_class = DiscountError
         error_type_field = "discount_errors"
+        webhook_events_info = [
+            WebhookEventInfo(
+                type=WebhookEventAsyncType.SALE_CREATED,
+                description=("A sale was created."),
```
Comment (IKarbowiak) [inline saleor/graphql/discount/mutations/sale/sale_create.py:78]: Same here 🚓

File: saleor/graphql/discount/mutations/sale/sale_delete.py [hunk 11]
```diff
@@ -25,6 +27,12 @@ class Meta:
         permissions = (DiscountPermissions.MANAGE_DISCOUNTS,)
         error_type_class = DiscountError
         error_type_field = "discount_errors"
+        webhook_events_info = [
+            WebhookEventInfo(
+                type=WebhookEventAsyncType.SALE_DELETED,
+                description=("A sale was deleted."),
```
Comment (IKarbowiak) [inline saleor/graphql/discount/mutations/sale/sale_delete.py:33]: And here

File: saleor/graphql/discount/mutations/sale/sale_remove_catalogues.py [hunk 12]
```diff
@@ -21,6 +23,12 @@ class Meta:
         permissions = (DiscountPermissions.MANAGE_DISCOUNTS,)
         error_type_class = DiscountError
         error_type_field = "discount_errors"
+        webhook_events_info = [
+            WebhookEventInfo(
+                type=WebhookEventAsyncType.SALE_UPDATED,
+                description=("A sale was updated."),
```
Comment (IKarbowiak) [inline saleor/graphql/discount/mutations/sale/sale_remove_catalogues.py:29]: And here

File: saleor/graphql/discount/mutations/sale/sale_update.py [hunk 13]
```diff
@@ -33,6 +35,16 @@ class Meta:
         permissions = (DiscountPermissions.MANAGE_DISCOUNTS,)
         error_type_class = DiscountError
         error_type_field = "discount_errors"
+        webhook_events_info = [
+            WebhookEventInfo(
+                type=WebhookEventAsyncType.SALE_UPDATED,
+                description=("A sale was updated."),
```
Comment (IKarbowiak) [inline saleor/graphql/discount/mutations/sale/sale_update.py:41]: And here

File: saleor/graphql/discount/mutations/sale/sale_update.py [hunk 14]
```diff
@@ -33,6 +35,16 @@ class Meta:
...
+            ),
+            WebhookEventInfo(
+                type=WebhookEventAsyncType.SALE_TOGGLE,
+                description=("Optionally triggered when a sale is started or stopped."),
```
Comment (IKarbowiak) [inline saleor/graphql/discount/mutations/sale/sale_update.py:45]: And here

**Review by IKarbowiak — APPROVED** [review]

**Review by maarcingebala — APPROVED** [review]

File: saleor/graphql/app/mutations/app_update.py [hunk 15]
```diff
@@ -29,6 +31,12 @@ class Meta:
         permissions = (AppPermission.MANAGE_APPS,)
         error_type_class = AppError
         error_type_field = "app_errors"
+        webhook_events_info = [
+            WebhookEventInfo(
+                type=WebhookEventAsyncType.APP_UPDATED,
+                description="An app was installed.",
```
[the remark this answers is not in the document]
Comment (korycins) [reply saleor/graphql/app/mutations/app_update.py:37]: 🚔  👮 
Thanks!

**Pull request merged**